src/shared/logs-show.c

```diff
@@ -1073,7 +1073,9 @@ int add_matches_for_unit(sd_journal *j, const char *unit) {
         );
 
         if (r == 0 && endswith(unit, ".slice")) {
-                char *m5 = strappend("_SYSTEMD_SLICE=", unit);
+                char *m5;
+
+                m5 = strjoina("_SYSTEMD_SLICE=", unit);
 
                 /* Show all messages belonging to a slice */
                 (void)(
```

Comment:
well, you wrote that code iirc... ;-)

Basically, I think you tried to follow the same coding style as used a few lines up for the more complex expressions. See 886a64fe. I think using the same style at both places is probably a good idea.